docs/content/docs/try-flink/write_flink_program_with_sql.md

```diff
@@ -0,0 +1,262 @@
+
+There are many ways to set up Flink but you will run it locally for the purpose of this tutorial. [Download Flink]({{< ref "docs/try-flink/local_installation#downloading-flink" >}}) and [start a local cluster]({{< ref "docs/try-flink/local_installation#starting-and-stopping-a-local-cluster" >}}) with one worker (the TaskManager).  
+
+The scripts for the SQL client are located in the `/bin` directory of Flink. You can start the client by executing:
+
+```sh
+./bin/sql-client.sh
+```
+
+You should see something like this:
+
+{{< img src="/fig/try-flink/flink-sql.png" alt="Flink SQL client" >}}
+
+
+## Step 2: Set up a data source with flink-faker
+
+Like with any Flink program, you will need a data source to connect to so that Flink can process it. There are many popular data sources but for the interest of this tutorial, you will be using [flink-faker](https://github.com/knaufk/flink-faker).  This custom [table source]({{< ref "docs/connectors/table/overview" >}}) is based on [Java Faker](https://github.com/DiUS/java-faker) and can generate fake data continuously in memory and in a realistic format. 
+
+Java Faker is a tool for generating this data and flink-faker exposes that as a source in Flink by implementing the [DynamicTableSource interface]({{< javadoc file="org/apache/flink/table/connector/source/DynamicTableSource.html" name="DynamicTableSource">}}). The dynamic table source has the logic of how to create a table source (in this case, from flink-faker), and then by adding a [factory]({{< javadoc file="org/apache/flink/table/factories/DynamicTableSourceFactory.html" name="DynamicTableSourceFactory">}}) for it you can expose it in the SQL API by referencing it with `"connector" = "faker"`.
+
+
+
+The next step is to make `flink-faker` discoverable by the Flink SQL client by following these [instructions](https://github.com/knaufk/flink-faker#adding-flink-faker-to-flink-sql-client).
+
+Once you have done that, create a table using this table source to confirm that the factory is loaded by executing the following query in the SQL client: 
+
+```sql
+CREATE TABLE test WITH ('connector' = 'faker');
+```
+
+If you see `[INFO] Execute statement succeed.`, then the table source has been loaded correctly.
+
+You are ready to start writing your first Flink program with SQL. 
+
+
+## Step 3: Consume the data via SQL
+
+For this tutorial, you are going to create a table that models a [Twitch](https://www.twitch.tv) gaming stream. This table will contain the following fields: user_name, game_name,viewer_count, started_at, location, and a timestamp.
+
+Use the [DDL syntax](https://www.geeksforgeeks.org/sql-ddl-dql-dml-dcl-tcl-commands/) `CREATE TABLE` to create this table containing these fields. You will also use the `WITH` clause to configure the connector (i.e. flink-faker). 
```

Comment:
we should link to other websites. the links get out of date quickly and require maintenance overhead for us. we should (and also have) our own DDL page for linking.